trio/_subprocess_platform/waitid.py

```diff
@@ -8,9 +8,14 @@
 from .._sync import CapacityLimiter, Event
 from .._threads import to_thread_run_sync
 
-assert (sys.platform != "win32" and sys.platform != "darwin") or not TYPE_CHECKING
+if TYPE_CHECKING:
+
+    def sync_wait_reapable(pid: int) -> None:
```

Comment:
Mind leaving a comment about why this function specifically is up here?

Reply:
I think it's to provide a definition if imported on other platforms.

Reply:
It's for `trio/_tests/test_subprocess.py:test_waitid_eintr()`:
```python
@slow
def test_waitid_eintr() -> None:
    # This only matters on PyPy (where we're coding EINTR handling
    # ourselves) but the test works on all waitid platforms.
    from .._subprocess_platform import wait_child_exiting

    if TYPE_CHECKING and sys.platform == "win32":
        return

    if not wait_child_exiting.__module__.endswith("waitid"):
        pytest.skip("waitid only")
    from .._subprocess_platform.waitid import sync_wait_reapable

    ...
```

where despite having a
`if TYPE_CHECKING and sys.platform == "win32": return` it will still complain about the import on line 522. I guess this one could be reported upstream?

Though I realize now it's probably better to silence the error in the test than to make people think they can import `sync_wait_reapable` globally on all platforms only to get an `ImportError` at runtime.